sale_coupon_product_management/models/sale_coupon_program.py

```diff
@@ -69,19 +69,24 @@ def _check_program_options(self):
 
     def _check_no_product_duplicate(self):
         for rec in self:
```

Comment:
```suggestion
        for rec in self.filtered("discount_line_product_id"):
```
This way you don't have to change the rest of the constrain method :smile:

Reply:
Thanks, done!